API/hermes_abi/hermes_abi.h

```diff
@@ -0,0 +1,400 @@
+  /// guaranteed to not be another exception. This should be called exactly once
+  /// after an exception is thrown.
+  HermesABIValue (*get_and_clear_js_error_value)(HermesABIContext *);
+  HermesABIASCIIRef (*get_native_exception_message)(HermesABIContext *);
+  /// Clear the stored native exception message. This should be called exactly
+  /// once after the message is retrieved.
+  void (*clear_native_exception_message)(HermesABIContext *);
+
+  /// Set the current error before returning control to the ABI. These are
+  /// intended to be used to throw exceptions from HostFunctions and
+  /// HostObjects.
+  void (*set_js_error_value)(HermesABIContext *, const HermesABIValue *);
+  void (
+      *set_native_exception_message)(HermesABIContext *, const char *, size_t);
+
+  HermesABIPropNameID (
+      *clone_prop_name_id)(HermesABIContext *, HermesABIPropNameID);
+  HermesABIString (*clone_string)(HermesABIContext *, HermesABIString);
+  HermesABISymbol (*clone_symbol)(HermesABIContext *, HermesABISymbol);
+  HermesABIObject (*clone_object)(HermesABIContext *, HermesABIObject);
+  HermesABIBigInt (*clone_big_int)(HermesABIContext *, HermesABIBigInt);
+
+  /// Check if the given buffer contains Hermes bytecode.
+  bool (*is_hermes_bytecode)(const uint8_t *, size_t);
+
+  /// Evaluate the given JavaScript buffer with an associated source URL in the
+  /// given context, and return the result.
+  HermesABIValueOrError (*evaluate_javascript)(
+      HermesABIContext *,
+      HermesABIBuffer *,
+      const char *,
+      size_t);
+
+  HermesABIObject (*get_global_object)(HermesABIContext *);
+  HermesABIStringOrError (
+      *create_string_from_ascii)(HermesABIContext *, const char *, size_t);
+  HermesABIStringOrError (
+      *create_string_from_utf8)(HermesABIContext *, const uint8_t *, size_t);
+
+  HermesABIObjectOrError (*create_object)(HermesABIContext *);
+
+  HermesABIBoolOrError (*has_object_property_from_string)(
+      HermesABIContext *,
+      HermesABIObject,
+      HermesABIString);
+  HermesABIBoolOrError (*has_object_property_from_prop_name_id)(
+      HermesABIContext *,
+      HermesABIObject,
+      HermesABIPropNameID);
+
+  HermesABIValueOrError (*get_object_property_from_string)(
+      HermesABIContext *,
+      HermesABIObject,
+      HermesABIString);
+  HermesABIValueOrError (*get_object_property_from_prop_name_id)(
+      HermesABIContext *,
+      HermesABIObject,
+      HermesABIPropNameID);
+  HermesABIVoidOrError (*set_object_property_from_string)(
+      HermesABIContext *,
+      HermesABIObject,
+      HermesABIString,
+      const HermesABIValue *);
+  HermesABIVoidOrError (*set_object_property_from_prop_name_id)(
+      HermesABIContext *,
+      HermesABIObject,
+      HermesABIPropNameID,
+      const HermesABIValue *);
+
+  HermesABIArrayOrError (
+      *get_object_property_names)(HermesABIContext *, HermesABIObject);
+
+  HermesABIArrayOrError (*create_array)(HermesABIContext *, size_t);
```

Comment:
I wonder how important is the passing of the array size?
In V8 creating an array with a known size produces the most inefficient array representation.

Reply:
Arrays in Hermes have far fewer representations than in V8, so that is less of a concern. The primary purpose of this API is to efficiently and conveniently provide this functionality on `jsi::Runtime`